pkg/k8s/labels.go

```diff
@@ -478,10 +478,17 @@ func IsMeshed(pod *corev1.Pod, controllerNS string) bool {
 	return pod.Labels[ControllerNSLabel] == controllerNS
 }
 
-// IsTapDisabled returns true if the pod has an annotation for explicitly
-// disabling tap
-func IsTapDisabled(pod *corev1.Pod) bool {
-	if valStr := pod.Annotations[ProxyDisableTapAnnotation]; valStr != "" {
+// IsTapDisabled returns true if a namespace or pod has an annotation for
+// explicitly disabling tap
+func IsTapDisabled(obj interface{}) bool {
+	var valStr string
+	switch resource := obj.(type) {
+	case *corev1.Pod:
+		valStr = resource.GetAnnotations()[ProxyDisableTapAnnotation]
```

Comment:
Hmmm, The outcome of both of these cases are same?

Reply:
Yep but in both cases `resource` is a now a type—namespace or pod—that has the `GetAnnotations` method.

Without this, `obj` is just some interface.

Reply:
Makes sense. Can we do [a case list](https://www.dotnetperls.com/switch-go) in that case so that we don't have to repeat that solution 🤔

Reply:
Ah right good call!

Reply:
This actually doesn't work. Compiler still gets an error `resource.GetAnnotations undefined (type interface{} has no field or method GetAnnotations)`